Tests/StreamChatTests/StateLayer/ChannelList_Tests.swift

```diff
@@ -0,0 +1,293 @@
+        env.channelListUpdaterMock.update_completion_result = .failure(testError)
+        await XCTAssertAsyncFailure(try await channelList.loadNextChannels(), testError)
+    }
+    
+    // MARK: - Pagination and State
+    
+    func test_loadChannels_whenAPIRequestSucceeds_thenStateUpdates() async throws {
+        setUpChannelList(usesMockedChannelUpdater: false)
+        let pageSize = 2
+        let channelListPayload = makeMatchingChannelListPayload(channelCount: pageSize)
+        env.client.mockAPIClient.test_mockResponseResult(.success(channelListPayload))
+
+        let pagination = Pagination(pageSize: pageSize, offset: 0)
+        let result = try await channelList.loadChannels(with: pagination)
+        XCTAssertEqualIgnoringOrder(channelListPayload.channels.map(\.channel.cid.rawValue), result.map(\.cid.rawValue))
+        XCTAssertEqualIgnoringOrder(channelListPayload.channels.map(\.channel.cid.rawValue), channelList.state.channels.map(\.cid.rawValue))
+    }
+    
+    func test_loadNextChannels_whenAPIRequestSucceeds_thenStateUpdates() async throws {
+        // Initial DB state
+        let existingChannelListPayload = makeMatchingChannelListPayload(channelCount: 2)
+        try await env.client.mockDatabaseContainer.write { session in
+            session.saveChannelList(payload: existingChannelListPayload, query: self.channelList.query)
+        }
+        setUpChannelList(usesMockedChannelUpdater: false)
+        
+        // Load more channels
+        let nextChannelListPayload = makeMatchingChannelListPayload(channelCount: 3)
+        env.client.mockAPIClient.test_mockResponseResult(.success(nextChannelListPayload))
+        let result = try await channelList.loadNextChannels()
+        XCTAssertEqual(nextChannelListPayload.channels.map(\.channel.cid), result.map(\.cid))
+        // State should contain both the existing and next channels
+        let expectedChannels = existingChannelListPayload.channels + nextChannelListPayload.channels
+        XCTAssertEqualIgnoringOrder(expectedChannels.map(\.channel.cid.rawValue), channelList.state.channels.map(\.cid.rawValue))
+    }
+    
+    // MARK: - Observing the Core Data Store
+    
+    func test_observingLocalStore_whenStoreChanges_thenStateChanges() async throws {
+        let expectation = XCTestExpectation(description: "State changed")
+        let incomingChannelListPayload = makeMatchingChannelListPayload(channelCount: 2)
+        let cancellable = channelList.state.$channels
+            .dropFirst() // ignore initial
+            .sink { channels in
+                XCTAssertEqualIgnoringOrder(incomingChannelListPayload.channels.map(\.channel.cid.rawValue), channels.map(\.cid.rawValue))
+                expectation.fulfill()
+            }
+        try await env.client.mockDatabaseContainer.write { session in
+            session.saveChannelList(payload: incomingChannelListPayload, query: self.channelList.query)
+        }
+        #if swift(>=5.8)
```

Comment:
I think it's best if we just use the old `wait(for: expectation)` API. We shouldn't bring complexity to the tests with if statements, wdyt?

Reply:
discussed internally, disregard it for now